lib/hammer_cli_foreman/virtual_machine.rb

```diff
@@ -0,0 +1,78 @@
+
+module HammerCLIForeman
+
+  class VirtualMachine < HammerCLIForeman::Command
+
+    resource :compute_resources
+    command_name 'virtual_machine'
+    desc _("View and manage compute resource's virtual machines")
+
+
+    class AvailableVirtualMachinesCommand < HammerCLIForeman::ListCommand
+
+      resource :compute_resources, :available_virtual_machines
```

Comment:
If `resource` was specified in parent command/class, there is no need to specify it again.

Simple `action :available_virtual_machines` is enough. Applies for command definitions below as well.